### PR TITLE
Use FRCAPI For Team Datafeed

### PR DESCRIPTION
**Pull request by phil-lopreiato**
Some groundwork for #1162 
 - Add a FMSAPI parser for team details endpoint
 - When fetching team details, also call the fmsapi endpoint. We need to hit both fmsapi and the usfirst page because each contains data the other doesn't have. fmsapi doesn't have team website, and usfirst doesn't have robot name/district info. [FRCAPI Feature Request Submitted](https://usfirst.collab.net/sf/go/artf4060?nav=1&_pagenum=1&returnUrlKey=1428517900636)

 - fmsapi team details parser returns a tuple of three models: Team, DistrictTeam, and Robot. DistrictTeam represents a team's "home district" We can add the points and events that count for points to be fields of this model later on. Robot is a model that contains a team's robot name in a year

TODO: needs tests

### CODE REVIEW

**Comment by flyingcow93** [comment]
Does this include ordering the events returned in chronological order?

**Comment by phil-lopreiato** [comment]
@flyingcow93 it does not. This covers moving team data scraping to FMSAPI

File: datafeeds/parsers/fms_api/fms_api_team_details_parser.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        if teamData['districtCode']:
+            districtTeam = DistrictTeam(
+                team=ndb.Key(Team, team.key_name),
+                year=self.year,
+                district=DistrictType.abbrevs[teamData['districtCode'].lower()]
+            )
+
+        robot = None
+        if teamData['robotName']:
+            robot = Robot(
```
Comment (fangeugene) [inline datafeeds/parsers/fms_api/fms_api_team_details_parser.py]: add `id=Robot.renderKeyName(team_key, year)`, implement `renderKeyName()` in Robot.py, and call `renderKeyName()` from the `@property def key_name`. Take a look at match.py for examples.

File: datafeeds/parsers/fms_api/fms_api_team_details_parser.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            team_number=teamData['teamNumber'],
+            name=teamData['nameFull'],
+            nickname=teamData['nameShort'],
+            address=address,
+            rookie_year=teamData['rookieYear']
+        )
+
+        districtTeam = None
+        if teamData['districtCode']:
+            districtTeam = DistrictTeam(
```
Comment (fangeugene) [inline datafeeds/parsers/fms_api/fms_api_team_details_parser.py]: Do the same with `id=...` as mentioned below with Robot.

File: datafeeds/parsers/fms_api/fms_api_team_details_parser.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        team = Team(
+            team_number=teamData['teamNumber'],
+            name=teamData['nameFull'],
+            nickname=teamData['nameShort'],
+            address=address,
+            rookie_year=teamData['rookieYear']
+        )
+
+        districtTeam = None
+        districtAbbrev = DistrictType.abbrevs[teamData['districtCode'].lower()]
```
Comment (fangeugene) [inline datafeeds/parsers/fms_api/fms_api_team_details_parser.py]: This line errors if team is not part of district.

File: tests/test_fms_api_team_parser.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+from google.appengine.ext import ndb
+from google.appengine.ext import testbed
+
+from consts.district_type import DistrictType
+from models.district_team import DistrictTeam
+from models.robot import Robot
+from models.team import Team
+
+
+class TestFMSAPITeamParser(unittest2.TestCase):
```
Comment (fangeugene) [inline tests/test_fms_api_team_parser.py]: Should add a test for team with no district

File: datafeeds/parsers/fms_api/fms_api_team_details_parser.py (hunk 3 above)
Comment (phil-lopreiato) [inline datafeeds/parsers/fms_api/fms_api_team_details_parser.py]: Good catch! That should be a line down

File: tests/test_fms_api_team_parser.py (hunk 4 above)
Comment (phil-lopreiato) [inline tests/test_fms_api_team_parser.py]: Added!

**Comment by phil-lopreiato** [comment]
FMSAPI feature request accepted :D 
https://usfirst.collab.net/sf/go/artf4060

**Comment by synth3tk** [comment]
:+1:

**Pull request merged**